logconsumer_test.go

```diff
@@ -31,7 +42,41 @@ func (g *TestLogConsumer) Accept(l Log) {
 		return
 	}
 
-	g.Msgs = append(g.Msgs, s)
+	g.mu.Lock()
```

Comment:
Could we add a comment here explaining why the synchronisation must be done at the test level? You already added it in #947 :)